src/Files.App/Actions/DebounceAction.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Files.App.Actions
+{
+	public abstract class DebouncedAction : IAction
+	{
+		private DateTime lastExecuted = DateTime.MinValue;
+		private readonly TimeSpan debounceTime;
+
+		protected DebouncedAction(TimeSpan debounceDuration)
+		{
+			debounceTime = debounceDuration;
+		}
+
+		public virtual bool CanExecuteNow()
+		{
+			return DateTime.Now - lastExecuted > debounceTime;
+		}
+
+		public virtual void MarkAsExecuted()
+		{
+			lastExecuted = DateTime.Now;
+		}
+
+		public abstract string Label { get; }
+		public abstract string Description { get; }
+		public abstract RichGlyph Glyph { get; }
+		public abstract HotKey HotKey { get; }
+		public abstract HotKey SecondHotKey { get; }
+		public abstract HotKey ThirdHotKey { get; }
+		public abstract HotKey MediaHotKey { get; }
+		public abstract bool IsExecutable { get; }
```

Comment:
These should be move above the constructor to keep the structure consistency. (See Class Structure in [the coding style](https://files.community/docs/contributing/code-style )if you're curious 😄 )